FuelLabs/fuel-indexer
# fix: support contract JSON ABI with callpaths

Closes #1320.

### Description

Adds support for JSON ABIs that have callpaths in the `type` field.

### Testing steps

The `fuel-indexer-test` contract ABI has been re-generated with `forc build --json-abi-with-callpaths` and CI should pass.

#### Manual testing

On `develop`:
- Go into any of the example contracts and run `forc build` to get a regular JSON ABI. Attempt to build the corresponding indexer for that contract. It should succeed. 🟢 
-  Run `forc build --json-abi-with-callpaths` to generate a JSON ABI with callpaths included. Attempt to build the corresponding indexer for that contract. It should fail. 🔴 

On this branch:
- Go into any of the example contracts and run `forc build` to get a regular JSON ABI. Attempt to build the corresponding indexer for that contract. It should succeed. 🟢 
-  Run `forc build --json-abi-with-callpaths` to generate a JSON ABI with callpaths included. Attempt to build the corresponding indexer for that contract. It should also succeed. 🟢
